src/Microsoft.Data.Sqlite/SqliteConnection.cs

```diff
@@ -94,12 +111,12 @@ public override void Open()
                 throw new InvalidOperationException(Strings.OpenRequiresSetConnectionString);
             }
 
-            var flags = Constants.SQLITE_OPEN_READWRITE | Constants.SQLITE_OPEN_CREATE;
+            var flags = Constants.SQLITE_OPEN_READWRITE | Constants.SQLITE_OPEN_CREATE | Constants.SQLITE_OPEN_URI;
             flags |= (ConnectionStringBuilder.Cache == SqliteConnectionCacheMode.Shared) ? Constants.SQLITE_OPEN_SHAREDCACHE : Constants.SQLITE_OPEN_PRIVATECACHE;
 
             var path = ConnectionStringBuilder.DataSource;
 
-            if (!path.Equals(":memory:", StringComparison.OrdinalIgnoreCase))
+            if (!IsInMemory(path))
```

Comment:
I want to provide some better abstractions in this area. Connection strings that look like `Data Source=file:thedb?mode=memory; Cache=Shared;` are terrible.